### PR TITLE
Update graphql-config to ^3.2.0; fix breakages

### PR DESCRIPTION
**Pull request by dahukish**
## Description

Fixes security updates regarding `node-fetch` by updating `graphql-config` to vs 3 as this version strips `graphql-response` which ends up using `node-fetch`.

Once this PR is in: https://github.com/Shopify/quilt/pull/1880 I will rebase.

## Type of change

- [x] `graphql-config-utilities`  Minor: New feature (non-breaking change which adds functionality)
expected)
- [x] `graphql-typescript-definitions`  Patch
expected)
- [x] `graphql-validate-fixtures`  Patch
expected)

## 🎩 

CI passes in sewing-kit (with beta releases): https://github.com/Shopify/sewing-kit/pull/2649

## Checklist

- [ ] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

File: packages/graphql-config-utilities/src/config.ts [hunk 1]
```diff
@@ -31,31 +36,26 @@ export function resolveSchemaPath(
   // schemaPath is nullable in graphq-config even though it cannot actually be
   // omitted. This function simplifies access to the schemaPath without
   // requiring a type guard.
```
Comment (atesgoral) [inline packages/graphql-config-utilities/src/config.ts:38]:
```suggestion
  // schema is nullable in graphql-config even though it cannot actually be
  // omitted. This function simplifies access to the schema without
  // requiring a type guard.
```

File: packages/graphql-config-utilities/src/config.ts [hunk 2]
```diff
@@ -31,31 +36,26 @@ export function resolveSchemaPath(
...
-  if (!project.schemaPath) {
+  if (!project.schema) {
     // this case should never happen with a properly formatted config file.
     // graphql-config currently does not perform any validation so it's possible
     // for a mal-formed schema to be loaded at runtime.
-    throw new Error(
-      `Missing GraphQL schemaPath for project '${resolveProjectName(project)}'`,
-    );
+    throw new Error(`Missing GraphQL schemaPath for project '${project.name}'`);
```
Comment (atesgoral) [inline packages/graphql-config-utilities/src/config.ts:43]:
```suggestion
    throw new Error(`Missing GraphQL schema for project '${project.name}'`);
```

File: packages/graphql-config-utilities/src/config.ts [hunk 3]
```diff
@@ -31,31 +36,26 @@ export function resolveSchemaPath(
...
-    const forProject = project.projectName
-      ? ` for project '${project.projectName}'`
-      : '';
+    const forProject = project.name ? ` for project '${project.name}'` : '';
     throw new Error(
       [
         `Schema not found${forProject}.`,
         `Expected to find the schema at '${schemaPath}' but the path does not exist.`,
-        `Check '${project.configPath}' and verify that schemaPath is configured correctly${forProject}.`,
+        `Check '${project.filepath}' and verify that schemaPath is configured correctly${forProject}.`,
```
Comment (atesgoral) [inline packages/graphql-config-utilities/src/config.ts:58]:
```suggestion
        `Check '${project.filepath}' and verify that schema is configured correctly${forProject}.`,
```

I'm making these suggestions without knowing we still retain "schemaPath" at the config level.

File: packages/graphql-config-utilities/src/tests/fixtures/error/.graphqlconfig [hunk 4]
```diff
@@ -0,0 +1 @@
+{ "name": "error-test", schemaPath: undefined}
```
Comment (atesgoral) [inline packages/graphql-config-utilities/src/tests/fixtures/error/.graphqlconfig:1]: Is this supposed to be valid JSON? :)

File: packages/graphql-config-utilities/src/config.ts (hunk 3 above)
Comment (dahukish) [reply packages/graphql-config-utilities/src/config.ts:58]: Yeah given the api changes of graphql-config 3 I think we might be able to drop it entirely. Some of the changes felt awkward. ie going out of my way not to break the api.

**Comment by dahukish** [comment]
I have 80+ linting issues to fix still

File: packages/graphql-config-utilities/src/tests/fixtures/error/.graphqlconfig (hunk 4 above)
Comment (dahukish) [reply packages/graphql-config-utilities/src/tests/fixtures/error/.graphqlconfig:1]: forgot to kill this whoops

File: packages/graphql-config-utilities/src/config.ts (hunk 3 above)
Comment (dahukish) [reply packages/graphql-config-utilities/src/config.ts:58]: `.graphqlconfig` can still have `schemaPath` this then becomes `schema` on the object instance.

File: packages/graphql-config-utilities/src/config.ts [hunk 5]
```diff
@@ -110,14 +114,40 @@ export function getGraphQLProjectForSchemaPath(
...
+        return (
+          resolve(project.dirpath, project.schema as string) === schemaPath
+        );
+      })
+      .shift() || config.projects[0];
+
+  if (
+    !project ||
+    !(project as GraphQLProjectConfig).schema ||
+    `${(project as GraphQLProjectConfig).dirpath}/${
```
Comment (atesgoral) [inline packages/graphql-config-utilities/src/config.ts:127]: Can't use `resolve` here as well?

File: packages/react-form/src/hooks/list/baselist.ts [hunk 6]
```diff
@@ -53,11 +53,9 @@ export function useBaseList<Item extends object>(
   validationDependencies: unknown[] = [],
 ): BaseList<Item> {
   const list = Array.isArray(listOrConfig) ? listOrConfig : listOrConfig.list;
-  const validates: FieldListConfig<Item>['validates'] = Array.isArray(
-    listOrConfig,
-  )
-    ? {}
-    : listOrConfig.validates || {};
+  const validates: FieldListConfig<Item>['validates'] = useMemo(() => {
```
Comment (atesgoral) [inline packages/react-form/src/hooks/list/baselist.ts:56]: Is this worth memoizing?

File: packages/graphql-config-utilities/src/config.ts [hunk 7]
```diff
@@ -64,22 +64,25 @@ export function resolveSchemaPath(
...
+  if (projects.length > 1) {
     // multi-project configuration, return an array of projects
-    return Object.values(projects);
+    return projects;
   }
 
-  const project = config.getProjectConfig();
-
-  if (project && project.schemaPath) {
+  if (projects[0] && projects[0]?.schema) {
```
[the remark this answers is not in the document]
Comment (dahukish) [reply packages/graphql-config-utilities/src/config.ts:79]: It was breaking due to the `?` before

File: packages/react-form/src/hooks/list/baselist.ts (hunk 6 above)
Comment (dahukish) [reply packages/react-form/src/hooks/list/baselist.ts:56]: linting

File: packages/graphql-config-utilities/src/config.ts (hunk 5 above)
Comment (dahukish) [reply packages/graphql-config-utilities/src/config.ts:127]: Yes?

File: package.json [hunk 8]
```diff
@@ -57,7 +57,7 @@
     "@sewing-kit/plugins": "^0.1.10",
     "@shopify/app-bridge": "^0.7.3",
     "@shopify/babel-preset": "^23.1.1",
-    "@shopify/eslint-plugin": "^36.0.2",
+    "@shopify/eslint-plugin": "^39.0.0",
```
Comment (vsumner) [inline package.json:60]: Could this be split out into a separate PR? I'm finding it difficult to split out the graphql changes from the lint changes.

Comment (dahukish) [reply package.json:60]: it is now https://github.com/Shopify/quilt/pull/1880/checks?check_run_id=2504490247
